src/Microsoft.AspNet.Mvc.ModelBinding/Binders/ArrayModelBinder.cs

```diff
@@ -19,9 +26,42 @@ public override Task<ModelBindingResult> BindModelAsync(ModelBindingContext bind
             return base.BindModelAsync(bindingContext);
         }
 
+        /// <inheritdoc />
         protected override object GetModel(IEnumerable<TElement> newCollection)
         {
+            if (newCollection == null)
+            {
+                return null;
+            }
+
             return newCollection.ToArray();
         }
+
+        /// <inheritdoc />
+        protected override void CopyToModel([NotNull] object target, IEnumerable<TElement> sourceCollection)
+        {
+            TElement[] targetArray = target as TElement[];
+            Debug.Assert(targetArray != null); // This binder is instantiated only for array model types.
+
+            if (sourceCollection != null && targetArray != null)
+            {
+                int maxIndex = targetArray.Length - 1;
+                int index = 0;
+                foreach (var element in sourceCollection)
+                {
+                    if (index > maxIndex)
+                    {
+                        break;
```

Comment:
what happens to the remaining elements in source array? what is the MVC 5.0 behavior in that case?